### PR TITLE
[SYCL][L0] Add proper handling of queue indexing

### PR DESCRIPTION
**Pull request by marekkozl**
* Fix spec violation while using custom queue index.

https://spec.oneapi.io/level-zero/latest/core/api.html#_CPPv4N23ze_command_queue_desc_t5indexE

### CODE REVIEW

File: sycl/plugins/level_zero/pi_level_zero.cpp [hunk 1]
```diff
@@ -1457,6 +1457,7 @@ _pi_queue::pi_queue_group_t::getZeQueue(uint32_t *QueueGroupOrdinal) {
   ZeCommandQueueDesc.ordinal = *QueueGroupOrdinal;
   ZeCommandQueueDesc.index = ZeCommandQueueIndex;
   ZeCommandQueueDesc.mode = ZE_COMMAND_QUEUE_MODE_ASYNCHRONOUS;
+  ZeCommandQueueDesc.flags = ZE_COMMAND_QUEUE_FLAG_EXPLICIT_ONLY;
```
Comment (smaslov-intel) [inline sycl/plugins/level_zero/pi_level_zero.cpp:1460]: How is this going to affect performance? Should we only set this for "index" != 0 and allow distributing across all engines for submissions to "index" == 0?

Comment (marekkozl) [reply sycl/plugins/level_zero/pi_level_zero.cpp:1460]: I know for a fact that this is a spec recommendation: `this flag should be used when applications want full control over multi-engine submission and scheduling.` Implementation details about performance gains/losses with experimental settings should be asked seperately as they aren't explicitly defined.

Comment (smaslov-intel) [reply sycl/plugins/level_zero/pi_level_zero.cpp:1460]: The thing is that we don't know if "we want full control" without perf data. I suggest only set this flag where it is mandatory, and not set for "0" index. And add a TODO comment o evaluate performance of having this flag there as well.

Comment (marekkozl) [reply sycl/plugins/level_zero/pi_level_zero.cpp]: This code segment already has a `full control` as it explicitly picks the engine. I have resolved your concerns and for time being this commit is just a temporary solution for functional spec violation.

**Review by smaslov-intel — APPROVED** [review]

**Review by smaslov-intel — APPROVED** [review]

**Pull request merged**